File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/adapter/holders/ViewHolderFactory.java
package com.mdgd.pokemon.ui.pokemon.adapter.holders;

import android.view.ViewGroup;

import com.mdgd.pokemon.ui.pokemon.adapter.PokemonPropertyViewHolder;
import com.mdgd.pokemon.ui.pokemon.infra.PokemonProperty;

public interface ViewHolderFactory {
    PokemonPropertyViewHolder<? extends PokemonProperty> create(ViewGroup parent);
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/adapter/holders/ViewHolderFactory.kt
package com.mdgd.pokemon.ui.pokemon.adapter.holders

import android.view.ViewGroup
import com.mdgd.pokemon.ui.pokemon.adapter.PokemonPropertyViewHolder
import com.mdgd.pokemon.ui.pokemon.infra.PokemonProperty

interface ViewHolderFactory {
    fun create(parent: ViewGroup?): PokemonPropertyViewHolder<out PokemonProperty>
}